File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlBetweenValue.cs
namespace T1.SqlSharp.Expressions;

public class SqlBetweenValue : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.BetweenValue;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_BetweenValue(this);
    }
    public required ISqlExpression Start { get; set; }
    public required ISqlExpression End { get; set; }
    public string ToSql()
    {
        return $"BETWEEN {Start.ToSql()} AND {End.ToSql()}";
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlBetweenValue.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlBetweenValue implements ISqlExpression {
    SqlType: SqlType = SqlType.BetweenValue;
    Span: TextSpan = new TextSpan();
    Start!: ISqlExpression;
    End!: ISqlExpression;

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_BetweenValue(this);
    }

    ToSql(): string {
        return `BETWEEN ${this.Start.ToSql()} AND ${this.End.ToSql()}`;
    }
} 